openff/qcsubmit/_tests/test_datasets.py

```diff
@@ -343,6 +343,38 @@ def test_componentresult_deduplication_iso():
     assert result.n_conformers == 2
 
 
+def test_componentresult_deduplication_tautomer():
+    """Reproduce issue #260 using molecules in the reported issue."""
+
+    result = ComponentResult(
+        component_name="Test iso deduplication",
+        component_description={},
+        component_provenance={},
+    )
+
+    # these two molecules have the same InChI key but differ in aromaticity ...
+    mol1 = Molecule.from_smiles(
+        smiles="[H]c1c(c(c(c2c1N(C(=C2C3=C(N(C(=O)C(=C3[H])[H])C([H])([H])C([H])([H])[H])[H])[H])[H])[H])F)[H]",
+        hydrogens_are_explicit=True,
+    )
+    mol1.generate_conformers(n_conformers=1)
+
+    mol2 = Molecule.from_smiles(
+        smiles="[H]c1c(c([n+](c(c1C2=C(N(c3c2c(c(c(c3[H])[H])F)[H])[H])[H])[H])C([H])([H])C([H])([H])[H])[O-])[H]",
+        hydrogens_are_explicit=True,
+    )
+    mol2.generate_conformers(n_conformers=1)
+
+    # this function returns a boolean equal to "is this molecule already in this dataset"
+    assert not result.add_molecule(mol1)
+    assert not result.add_molecule(mol2)
+
+    # ... and so the expected behavior is that they're deduplicated into the same
+    # "molecule" but the separate `Molecule` objects exist as distinct conformers
+    assert result.n_molecules == 1
+    assert result.n_conformers == 2
```

Comment:
I do not understand how this class and method are meant to behave, so everything I've written here should be fact-checked. This test is mostly cribbed from `test_componentresult_deduplication_iso ` above and the shifting boundaries of molecule identity make it hard to understand (at least for me at this particular moment). Brownie points if `ComponentResult.add_molecule` can be updated in a way that I understand what's going on

Reply:
I agree with you assessment that the second call to `add_molecule` should return `True`. Working on updating that now.